modules/post/windows/gather/checkvm.rb

```diff
@@ -180,6 +208,59 @@ def qemu?
     key_path = 'HKLM\HARDWARE\DESCRIPTION\System\CentralProcessor\0'
     return true if registry_getvaldata(key_path, 'ProcessorNameString') =~ /qemu/i
 
+    return true if registry_getvaldata('HKLM\HARDWARE\DESCRIPTION\System', 'SystemBiosVersion') =~ /qemu/i
```

Comment:
`HKLM\HARDWARE\DESCRIPTION\System\SystemBiosVersion` is a Multi String value and this call returns an array of characters:
```
[4] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> registry_getvaldata('HKLM\HARDWARE\DESCRIPTION\System', 'SystemBiosVersion')
=> ["I",
 "N",
 "T",
 "E",
 "L",
 " ",
 " ",
 "-",
...
```

So, this regex won't work. Maybe this would be enough to fix it (not tested):
```suggestion
    return true if registry_getvaldata('HKLM\HARDWARE\DESCRIPTION\System', 'SystemBiosVersion').join =~ /qemu/i
```

I also noticed this issue in many times in this file and they should also be fixed.